exporter/exporterhelper/common.go

```diff
@@ -101,19 +101,11 @@ func fromOptions(options []Option) *baseSettings {
 // Option apply changes to baseSettings.
 type Option func(*baseSettings)
 
-// WithStart overrides the default Start function for an exporter.
-// The default shutdown function does nothing and always returns nil.
-func WithStart(start componenthelper.Start) Option {
+// WithComponentOptions adds options that apply to a generic component
+// (such as start and shutdown hooks).
+func WithComponentOptions(opts ...componenthelper.Option) Option {
 	return func(o *baseSettings) {
-		o.componentOptions = append(o.componentOptions, componenthelper.WithStart(start))
-	}
-}
-
-// WithShutdown overrides the default Shutdown function for an exporter.
-// The default shutdown function does nothing and always returns nil.
-func WithShutdown(shutdown componenthelper.Shutdown) Option {
-	return func(o *baseSettings) {
-		o.componentOptions = append(o.componentOptions, componenthelper.WithShutdown(shutdown))
+		o.componentOptions = append(o.componentOptions, opts...)
```

Comment:
Do we want to call append here or overwrite?

Reply:
I chose append on purpose, but I don't have a strong preference.

Reply:
what was the reasoaning?

Reply:
If a piece of code configures startup and shutdown options far apart for any reason, the append approach gives them a way forward (because they can replace `WithStart -> WithComponentOptions(c.WithStart)` etc.) rather than forcing them to pull the two calls close together. As I said, it's not a big deal either way.